samples/Shell/Fabimals/Fabimals/Components/AnimalListPage.fs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018-2019 Fabulous contributors. See LICENSE.md for license.
+namespace Fabimals.Components
+
+open Fabulous.DynamicViews
+open Fabimals.Models
+open Fabimals.Components
+open Xamarin.Forms
+
+module AnimalList =
+    type Msg =
+        | SearchHandlerMsg of SearchHandlers.Msg
+        | SelectAnimal of Animal
+        
+    type CmdMsg = NoOp
+        
+    type ExternalMsg =
+        | NavigateToDetails of Animal
+
+    type Model =
+        { PageTitle: string
+          IsTopBarDisplayed: bool
+          AllAnimals: Animal list
+          FilteredAnimals: Animal list }
+
+    let init title isTopBarDisplayed data =
+        { PageTitle = title; IsTopBarDisplayed = isTopBarDisplayed; AllAnimals = data; FilteredAnimals = [] }
+
+    let update msg model : Model * CmdMsg list * ExternalMsg list =
+        match msg with
+        | SearchHandlerMsg (SearchHandlers.Msg.QueryChanged newValue) ->
+            let filteredAnimals = model.AllAnimals |> List.filter (fun a -> a.Name.ToLower().Contains(newValue.ToLower()))
+            { model with FilteredAnimals = filteredAnimals }, [], []
+        | SearchHandlerMsg (SearchHandlers.Msg.AnimalSelected animal) ->
+            model, [], [(NavigateToDetails animal)]
+        | SelectAnimal animal ->
+            model, [], [(NavigateToDetails animal)]
+
+    let navigateToAfterSelectionChanged dispatch (args: SelectionChangedEventArgs) =
+        match args.CurrentSelection |> Seq.tryHead with
+        | None -> ()
+        | Some item ->
+            let data = item :?> ItemListElementData
+            let animal = data.Key.GetAttributeKeyed(ViewAttributes.TagAttribKey) :?> Animal
+            dispatch (SelectAnimal animal)
+
+    let view model dispatch =
+        dependsOn (model.PageTitle, model.IsTopBarDisplayed, model.AllAnimals, model.FilteredAnimals) (fun model (pageTitle, isTopBarDisplayed, allAnimals, filteredAnimals) ->
```

Comment:
you could replace `model` with an underscore. model is not used in there.

Reply:
Here `model` is of type `DoNotUseModelInsideDependsOn`
It needs to match the name `model` to prevent access to the real model inside `dependsOn .
https://github.com/fsprojects/Fabulous/blob/dcf5dbe175fa617977a6aa08a5a25dff25f5c292/src/Fabulous.Core/ViewHelpers.fs#L24-L30

Reply:
Alright, didn't know about that.